component.js

```diff
@@ -255,9 +259,21 @@ function factory (options) {
     }
 
     if (!hasShouldComponentUpdate(mixins)) {
-      mixins = [{
+      mixins.unshift({
         shouldComponentUpdate: _shouldComponentUpdate
-      }].concat(mixins);
+      });
+    }
+
+    if (!hasComponentWillReceiveProps(mixins)) {
```

Comment:
Agreed. Should be able to have multiple `componentWillUpdate` and `componentWillReceiveProps` mixins. And I don't think they should be overridable.